src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperContextExtensions.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Framework.Logging;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.TagHelpers;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Utility related extensions for <see cref="TagHelperContext"/>.
+    /// </summary>
+    public static class TagHelperContextExtensions
+    {
+        /// <summary>
+        /// Determines whether a <see cref="ITagHelper" /> should run based on the presence of attributes it requires
+        /// to be present, non null, non empty, and non whitepsace.
+        /// </summary>
+        /// <param name="context">The <see cref="TagHelperContext"/>.</param>
+        /// <param name="requiredAttributes">The attributes the <see cref="ITagHelper" /> requires in order to run.</param>
+        /// <param name="logger">An optional <see cref="ILogger"/> to log warning details to.</param>
+        /// <returns>A <see cref="bool"/> indicating whether the <see cref="ITagHelper" /> should run.</returns> 
+        public static bool ShouldProcess([NotNull]this TagHelperContext context, [NotNull]IEnumerable<string> requiredAttributes, ILogger logger = null)
```

Comment:
pls wrap long lines and rename to something slightly less generic e.g. `AreAllRequiredAttributesPresent`